From 515a4d0088b06a32ef5157ef0c8bbb8c524e2533 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Thu, 27 Jun 2024 17:08:25 +0200
Subject: [PATCH 0955/1695] dts: stm32: declare RISAB instances as firewall
 controllers on stm32mp23

Declare RISABs as access controllers so that it's possible to check
firewall access rights on address ranges, and reconfigure dynamically
RISAB regions.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git f1925be50f432d15d2ee6828ebb07d3e81fb358f dts: stm32: declare RISAB instances as firewall controllers on stm32mp23]
Change-Id: I8a9f59dee846afd82120bccd68a46cf02d40ed9b
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/390284
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 core/arch/arm/dts/stm32mp231.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp231.dtsi b/core/arch/arm/dts/stm32mp231.dtsi
index 2adcbe36d..0e500a2bd 100644
--- a/core/arch/arm/dts/stm32mp231.dtsi
+++ b/core/arch/arm/dts/stm32mp231.dtsi
@@ -598,6 +598,7 @@
 			reg = <0x420f0000 0x1000>;
 			clocks = <&rcc CK_ICN_LS_MCU>;
 			st,mem-map = <0xa000000 0x20000>;
+			#access-controller-cells = <1>;
 		};
 
 		risab2: risab@42100000 {
@@ -605,6 +606,7 @@
 			reg = <0x42100000 0x1000>;
 			clocks = <&rcc CK_ICN_LS_MCU>;
 			st,mem-map = <0xa020000 0x20000>;
+			#access-controller-cells = <1>;
 		};
 
 		risab3: risab@42110000 {
@@ -612,6 +614,7 @@
 			reg = <0x42110000 0x1000>;
 			clocks = <&rcc CK_ICN_LS_MCU>;
 			st,mem-map = <0xa040000 0x20000>;
+			#access-controller-cells = <1>;
 		};
 
 		risab4: risab@42120000 {
@@ -619,6 +622,7 @@
 			reg = <0x42120000 0x1000>;
 			clocks = <&rcc CK_ICN_LS_MCU>;
 			st,mem-map = <0xa060000 0x20000>;
+			#access-controller-cells = <1>;
 		};
 
 		risab5: risab@42130000 {
@@ -626,6 +630,7 @@
 			reg = <0x42130000 0x1000>;
 			clocks = <&rcc CK_ICN_LS_MCU>;
 			st,mem-map = <0xa080000 0x20000>;
+			#access-controller-cells = <1>;
 		};
 
 		risab6: risab@42140000 {
@@ -633,6 +638,7 @@
 			reg = <0x42140000 0x1000>;
 			clocks = <&rcc CK_ICN_LS_MCU>;
 			st,mem-map = <0xa0a0000 0x20000>;
+			#access-controller-cells = <1>;
 			status = "disabled";
 		};
 
-- 
2.47.2

